bigquery_etl/cli/query.py

```diff
@@ -1262,6 +1265,150 @@ def render(name, sql_dir, output_dir):
+    # Will be None if launched from Airflow.  Also ctx.args is not populated at this stage.
+    if partition_date:
+        parsed = _parse_partition_setting(partition_date)
+        if parsed is None:
+            raise click.BadParameter("Format must be <column-name>::<yyyy-mm-dd>")
+        return parsed
+    return None
+
+
+def _parse_check_output(output: str) -> str:
+    output = output.replace("\n", " ")
+    if "ETL Data Check Failed:" in output:
+        return f"ETL Data Check Failed:{output.split('ETL Data Check Failed:')[1]}"
+    return output
+
+
+@query.command(
+    help="""
+    \b
+    UNDER ACTIVE DEVELOPMENT See https://mozilla-hub.atlassian.net/browse/DENG-919
+
+    Run ETL checks.
+    \b
+
+    Example:
+
+     ./bqetl query check ga_derived.downloads_with_attribution_v2 --partition download_date::2023-05-01
+    """,
+    context_settings=dict(
+        ignore_unknown_options=True,
+        allow_extra_args=True,
+    ),
+)
+@click.argument("name")
+@sql_dir_option
+@click.option(
+    "--partition",
+    "-p",
+    help="Partition to check, format <column-name>::<yyy-mm-dd>, must be provided if not executing in Airflow ",
+    type=click.UNPROCESSED,
+    callback=_validate_partition_date,
+    required=False,
+)
+@click.pass_context
+def check(ctx, name, sql_dir, partition):
+    """Run a check."""
+    if not is_authenticated():
+        click.echo(
+            "Authentication to GCP required. Run `gcloud auth login` "
+            "and check that the project is set correctly."
+        )
+        sys.exit(1)
+
+    checks_file, project_id, dataset_id, table = paths_matching_checks_pattern(
+        name, sql_dir, project_id=None
+    )
+
+    _check_query(
+        checks_file,
+        project_id,
+        dataset_id,
+        table,
+        partition,
+        ctx.args,
+    )
+
+
+def _check_query(
+    checks_file,
+    project_id,
+    dataset_id,
+    table,
+    partition,
+    query_arguments,
+):
+    """Run the check."""
+    query_arguments.append("--use_legacy_sql=false")
+    if project_id is not None:
+        query_arguments.append(f"--project_id={project_id}")
+
+    # Partition will be None if from Airflow
+    if partition is None:
+        # Need to check the query_arguments for the value
+        for parameter in query_arguments:
+            if parameter.startswith("--parameter"):
+                param_value = parameter.split("=")[1]
+                partition = _parse_partition_setting(param_value)
+                # once we have a value that passed the date check stop checking.
+                if partition is not None:
+                    break
+    else:
+        # We have a partition value from the CLI so add to the query_arguments.
+        # There should only be 1
+        key, value = next(iter(partition.items()))
+        query_arguments.append(f"--parameter={key}::{value}")
+
+    if partition is None:
+        raise ValueError("No partition specified to check.")
```

Comment:
We do have some tables that are not partitioned, so we should support them as well

Reply:
can you provide an example table I can use for dev testing?

Reply:
https://github.com/mozilla/bigquery-etl/blob/505c895f624ce7f9b13fc3c16752f0d9a9420456/sql/moz-fx-data-marketing-prod/adjust_derived/firefox_mobile_installs_v1/query.sql for example.

Usually, unpartitioned (non-incremental) queries are marked in the metadata with `incremental: false`. There are a couple more if you need more examples

Reply:
I wonder if as part of the base work for this tool we should put some effort into ensuring all metadata follow at least some rules. I don't think currently all of them would have the `incremental` label so we can't really rely on it. Or maybe we could use something like the `date_partitioning_parameter` inside the `scheduling` [section](https://github.com/mozilla/bigquery-etl/blob/main/sql/moz-fx-data-shared-prod/apple_ads_external/ad_group_report_v1/metadata.yaml#L10) to automatically determine which field to use and whether check should be against the whole table or just a partition?

Reply:
I don't think that the Phase 1 implementation of BQETL checks is the place to enforce rules on the metadata.yaml files.  It should/could be a separate Issue in bigquery-etl itself.